tests/radio.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { test, expect } from '@playwright/test';
+import { waitForCustomEvent, waitForCustomEventTimeout } from './lib/helper.ts'
+
+
+test('Radio E2E Test, Options, Selected, Click, Keydown ', async ({ page }) => {
+  await page.goto('/test.html');
+
+  // Confirm All Options Render Correctly As Buttons
+  const radio = page.getByTestId("radio-selected-test")
+  await expect(radio).toBeVisible()
+
+  const opt1 = page.getByRole('button', { name: 'Opt 1' })
+  const opt2 = page.getByRole('button', { name: 'Opt 2' })
+  const opt3 = page.getByRole('button', { name: 'Opt 3' })
+
+  await expect(opt1).toBeVisible()
+  await expect(opt2).toBeVisible()
+  await expect(opt3).toBeVisible()
+
+  await expect(opt1).toHaveText('Opt 1')
+  await expect(opt2).toHaveText('Opt 2')
+  await expect(opt3).toHaveText('Opt 3')
+
+  // Confirm Selected Value - Opt 3
+  await expect(opt1).toHaveClass(/bg-white/)
+  await expect(opt2).toHaveClass(/bg-white/)
+  await expect(opt3).toHaveClass(/bg-gray-9/)
+
+  // Confirm Click Changes Selected to Opt 2
+  const opt2Selected = waitForCustomEvent(page,'input')
```

Comment:
btw, i just made a helper function for switch that tests the event dispatch and also the value dispatched, if you want to add that i can try to push that in first

Reply:
Nice!!!!  that would be great, thanks!

Reply:
https://github.com/viamrobotics/prime/pull/194